src/panels/lovelace/cards/hui-sensor-card.ts

```diff
@@ -69,34 +80,40 @@ const calcPoints = (
   max: number
 ): number[][] => {
   const coords = [] as number[][];
-  const margin = 5;
   const height = 80;
-  width -= 10;
   let yRatio = (max - min) / height;
   yRatio = yRatio !== 0 ? yRatio : height;
   let xRatio = width / (hours - (detail === 1 ? 1 : 0));
   xRatio = isFinite(xRatio) ? xRatio : width;
+
+  const first = history.filter(Boolean)[0];
+  let last = [average(first), lastValue(first)];
+
   const getCoords = (item, i, offset = 0, depth = 1) => {
     if (depth > 1) {
       return item.forEach((subItem, index) =>
         getCoords(subItem, i, index, depth - 1)
       );
     }
-    const average =
-      item.reduce((sum, entry) => sum + parseFloat(entry.state), 0) /
-      item.length;
 
-    const x = xRatio * (i + offset / 6) + margin;
-    const y = height - (average - min) / yRatio + margin * 2;
+    const x = xRatio * (i + offset / 6);
+
+    if (item) {
+      last = [average(item), lastValue(item)];
+    }
+    const y = height - ((item ? last[0] : last[1]) - min) / yRatio;
     return coords.push([x, y]);
   };
 
-  history.forEach((item, i) => getCoords(item, i, 0, detail));
+  for (let i = 0; i < history.length; i += 1) {
+    getCoords(history[i], i, 0, detail);
+  }
```

Comment:
Why this change?

Reply:
This is for when there isn't enough history to fill the hours requested. The previous method would start the line in the middle of the card if there were only 12 hours of data when 24 were requested

Reply:
Might be me, but I don't see a difference in what it does, just a more difficult way to write the same ;-)

Reply:
![image](https://user-images.githubusercontent.com/18730868/73752570-336c6c00-472f-11ea-9c72-49cbcc02189d.png)

The difference using the previous method. For loop starts at 0 and loops through no matter if the item is []. For Each ignores NULL/Undefined/[] items

Reply:
Therefore i starts at ex. 14. Meaning the first X value is 14 * xRatio